apps/prairielearn/src/lib/opsbot.ts

```diff
@@ -0,0 +1,72 @@
+import detectMocha from 'detect-mocha';
+
+import { makeWithData } from '@prairielearn/error';
+import { config } from './config';
+import { logger } from '@prairielearn/logger';
+import fetch, { Response } from 'node-fetch';
+
+export function canSendMessages() {
+  return detectMocha() || !!config.secretSlackOpsBotEndpoint;
+}
+
+export async function sendMessage(msg: string): Promise<Response | null> {
+  if (detectMocha()) {
+    return new Response('Dummy test body', { status: 200, statusText: 'OK' });
+  }
+
+  // No-op if there's no url specified
+  if (!config.secretSlackOpsBotEndpoint) {
+    return null;
+  }
+
+  return fetch(config.secretSlackOpsBotEndpoint, {
+    method: 'POST',
+    body: JSON.stringify({ text: msg }),
+    headers: { 'Content-Type': 'application/json' },
+  });
```

Comment:
As long as you're refactoring this code, consider making `sendMessage` match the behavior of `sendSlackMessage` and throw an error based on the status code instead of returning the raw response object.

Reply:
Done in 602b31d73cece40bce993ac2a55c8a31e12e16bc.